src/test/java/org/terasology/logic/inventory/ItemCommandsTest.java

```diff
@@ -0,0 +1,124 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.logic.inventory;
+
+import com.google.common.collect.Sets;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.TeraMath;
+import org.terasology.moduletestingenvironment.ModuleTestingEnvironment;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.block.items.BlockItemFactory;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Verify that the {@link ItemCommands} are working as expected.
+ */
+public class ItemCommandsTest {
+
+    static final String URI_DIRT = "CoreAssets:Dirt";
+
+    private static ModuleTestingEnvironment mte;
+
+    private Context hostContext;
+    private EntityManager entityManager;
+    private InventoryManager inventoryManager;
+    private BlockManager blockManager;
+    private ItemCommands itemCommands;
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        mte = new ModuleTestingEnvironment() {
+            @Override
+            public Set<String> getDependencies() {
+                return Sets.newHashSet("CoreAssets", "Inventory");
+            }
+        };
+        mte.setup();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        mte.tearDown();
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        hostContext = mte.getHostContext();
+        entityManager = hostContext.get(EntityManager.class);
+        inventoryManager = hostContext.get(InventoryManager.class);
+        blockManager = hostContext.get(BlockManager.class);
+        itemCommands = hostContext.get(ItemCommands.class);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {0, 1, 99, 100})
+    public void giveItemSingleBlockStack(int amount) {
+
+        Context clientContext = mte.createClient();
+        LocalPlayer localPlayer = clientContext.get(LocalPlayer.class);
+        final EntityRef player = localPlayer.getClientEntity();
+        final EntityRef character = localPlayer.getCharacterEntity();
+        character.upsertComponent(InventoryComponent.class, maybeInventory -> maybeInventory.orElse(new InventoryComponent(40)));
+
+        final BlockItemFactory factory = new BlockItemFactory(entityManager);
+        final EntityRef blockItem = factory.newInstance(blockManager.getBlockFamily(URI_DIRT), amount);
```

Comment:
I think you're doing this in #12 as well. We could consider to create a `TestUtils` class 🤔

Reply:
I think this not needed here... at least not as of now... will be tricky enough to figure out what classifies the added block to be "correct"